hack/verify-typecheck-dockerless.sh

```diff
@@ -0,0 +1,30 @@
+#!/usr/bin/env bash
+
+# Copyright 2018 The Kubernetes Authors.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+set -o errexit
+set -o nounset
+set -o pipefail
+
+KUBE_ROOT=$(dirname "${BASH_SOURCE[0]}")/..
+
+cd "${KUBE_ROOT}"
+# verify the dockerless build
+# @TODO(mattjmcnaughton) add the link to the KEP once its merged.
+# @TODO(mattjmcnaughton) We should not need to add a new `verify-typecheck-*`
```

Comment:
we probably don't want a proliferation of tags and the tags should be removed once the code they're gating is removed
(e.g. => complete cloud provider extraction means we can remove providerless, dockershim out of tree => same for this one)

when one of these fails the CI output tells you how to run that particular verify check locally, which is pretty useful.

it seems pretty likely that a change will break a particular tag and the context may help people unfamiliar with this odd internal detail understand what is going on.

Reply:
Just to confirm - are you arguing that this `TODO` doesn't make sense, and we should be fine with creating a separate verify script for each tag?

If yes, that works perfectly fine for me. Happy to delete this todo. Let me know if I'm misunderstanding you :)

Reply:
yeah, I  think we should just do one per tag and try not to create very many of these tags.

the end goal for these tags should be to eliminate them as the code they gate is actually deleted.